recorder.go

```diff
@@ -244,6 +244,23 @@ func GetLightStepAccessToken(lsTracer ot.Tracer) (string, error) {
 	}
 }
 
+type Closer interface {
+	Close() error
+}
+
+func CloseTracer(tracer ot.Tracer) error {
```

Comment:
The way this function is setup is not `lightstep` specific: any `basictracer.Tracer` that has a `Recorder` that can be closed will work here. So it feels to me that this generic function is odd-looking as part of this package. 

Maybe just make the `lightstep.Tracer` type implement `io.Closer`? In another piece of code where colleagues and I implement our own `Tracer` type (which originally motivated my #56), we made a wrapper interface type that clearly communicate that our `Tracer` can be closed:

```go
// Tracer is an opentracing.Tracer that can be closed.
type Tracer interface {
	io.Closer
	opentracing.Tracer
}

func New(ctx context.Context, addr string, opts ...Option) (Tracer, error) {
    // ...
}
```

In my opinion, this makes the API simpler to use and less error prone, since it only works on types that are meant to be worked with (versus doing runtime type assertions like here).